pymodbus/datastore/remote.py

```diff
@@ -38,25 +39,37 @@ def validate(self, _fc_as_hex, _address, _count):
         """
         return True
 
-    def getValues(self, fc_as_hex, _address, _count=1):
+    async def async_getValues(self, fc_as_hex, address, count=1):
         """Get values from real call in validate."""
         if fc_as_hex in self._write_fc:
             return [0]
         group_fx = self.decode(fc_as_hex)
         func_fc = self.__get_callbacks[group_fx]
-        self.result = func_fc(_address, _count)
+        kwargs = {}
+        if self.slave:
+            kwargs["slave"] = self.slave
+        self.result = await getattr(self._client, func_fc)(address, count, **kwargs)
```

Comment:
This looks wrong to me, the code is designed so we do not need to call getattr, which is very expensive function.